src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Models/AnalysisServicesClient.cs

```diff
@@ -236,6 +261,57 @@ public void ResumeServer(string resourceGroupName, string serverName)
             _client.Servers.Resume(resourceGroupName, serverName);
         }
 
+		public GatewayInfo GetGatewayInfo(string resourceId)
+		{
+			var availableGateways = getAvailableGateways();
+			foreach (var gateway in availableGateways.value)
+			{
+				if (gateway.id.Equals(resourceId))
+				{
+					return gateway;
+				}
+			}
+
+			return null;
+		}
+
+
+		private const string WebResourceProviderVersion = "api-version=2016-06-01";
+
+        private GatewayDetails getConnectionGateway(string Name)
+        {
+            var availableGateways = getAvailableGateways();
+            GatewayDetails targetGateway = null;
+            foreach( var gateway in availableGateways.value)
+            {
+                if (gateway.name.Equals(Name))
+                {
+                    targetGateway = new GatewayDetails(gateway.id);
+                }
+            }
+
+            return targetGateway;
+        }
+
+        private GatewayCollection getAvailableGateways()
+        {
+            var _baseUrl = _client.BaseUri.AbsoluteUri;
```

Comment:
why is this not part of the management client?

Reply:
This is under Microsoft.Web API version 2016-06-01, which did not have rest spec version available. 
Available versions are found in (https://github.com/Azure/azure-rest-api-specs/tree/master/specification/web/resource-manager/Microsoft.Web/stable).
This resource is linked resource, is not under Azure analysis services.

Reply:
Can you use an existing version?  If so, we can put this into common code

Reply:
@taiwu You need to use a version that has a REST spec available

Reply:
@taiwu ping on the comment above.

Reply:
@maddieclayton as mentioned in a comment below on Feb 6th, there are no rest spec api for this.

Reply:
@taiwu  The problem is, we don't know what the support deatils are for this api-version, so it is dangerous to include it in a cmdlet.  Why would you need data from such an old API that no longer exists?